### DISCUSSION TITLE
Support optimization for large memory devices

### DISCUSSION THREAD

**Discussion by shenyan-008** [immich-app/immich#13446] [Q&A]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [x] Yes

### The feature

I hope to support optimization for large memory devices. The access speed of 64GB memory is no different from that of 8GB memory, both are very slow./希望支持对大内存设备的优化，64GB内存的访问速度与8GB内存的访问速度没差别，都很慢

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by alextran1502** [comment]
Can you elaborate on your feature request, I don't understand what you are asking for

**Reply by shenyan-008** [reply]
Is the loading time too slow for gallery access? The gallery currently has nearly 700,000 pictures, but every time you close the page and reopen it, you have to wait for a while before it can be fully displayed. Especially for APP access, it takes nearly 30 seconds to display, and the APP will crash after a quick pull. This makes my experience very different from that of most gallery websites./对于图库的访问的加载是否过于缓慢？图库当前拥有接近70W张图片，但是每次关闭页面重新打开后都需要等待一段时间才能够显示完整，特别是APP的访问，需要等待接近30s才能够显示，并且快速拉动后APP会奔溃。这让我与大部分的图库网站的体验相差过大

**Reply by alextran1502** [reply]
Ah so this is optimization topics for larger library. Thanks for clarifying

**Comment by mertalev** [comment]
You can configure Postgres to use more RAM, keeping more of the library metadata in memory and making things much faster with larger libraries. The `shared_buffers` and `work_mem` settings are a good place to start. For example, you can set `shared_buffers` to 8GB and `work_mem` to 1GB.

Also, running the `VACUUM FULL;` command will clean the DB and bring it into optimal condition.
